### ISSUE TITLE
[BUG] github_branch_protection_v3 tries to delete the contexts every time

### ISSUE THREAD

**Issue by julioagos** [integrations/terraform-provider-github#1701]
### Problem
I created some branch protection rules for some branches. These rules included required checks. After I applied them, it created matching contexts and a new check that is just one of the required checks with an app_id that always changes (First 4 digits remain the same, the rest changes). Now, every time I ran plan, it says that it is going to delete them again. The branch protections are working just as intended but, running plan always generates meaningless noise that get's in the way especially since we have a lot of branch protection rules once we fully migrate it over to Terraform. The applies are all run by `Atlantis` on PRs. This is what an example output of what `terraform state show` looks like for one of the branches:
```
    required_status_checks {
        checks         = [
            "test 1",
            "test 2",
            "test 1:<random-numbers>",
        ]
        contexts       = [
            "test 1",
            "test 2",
        ]
    }
```
Running `terraform plan`  returns something like this
```
      ~ required_status_checks {
          ~ checks         = [
              - "test 1:<random-numbers>",
                # (2 unchanged elements hidden)
            ]
          ~ contexts       = [
              - "test 1",
              - "test 2",
            ]
            # (2 unchanged attributes hidden)
        }
```
There are no error messages at any point and the apply works as intended. It states that all plans have been applied.

### Versions
Terraform: v1.4.4
GitHub Provider: Originally 5.23.0 but also happens in 5.25.1

**Comment by svalevka** [comment]
We have a very similiar issue with 5.25.1 provider.


```
  # module.github_repository.github_branch_protection_v3.branch["redacted_name.master"] will be updated in-place
  ~ resource "github_branch_protection_v3" "branch" {
        id                              = "redacted_name:master"
        # (6 unchanged attributes hidden)

      ~ required_status_checks {
          ~ checks         = [
              - "Naming conventions / Pull request:0",
                # (1 unchanged element hidden)
            ]
          ~ contexts       = [
              - "Naming conventions / Pull request",
            ]
            # (2 unchanged attributes hidden)
        }
```

**Comment by 0x46616c6b** [comment]
I have the same problem. Since #1774 (v5.31.0) the `checks` diffs are gone but every plan produce a diff for `contexts` (regardless if they are set or not).

**Comment by 0x46616c6b** [comment]
You are correct; the `github_branch_protection` works better with `required_status_checks`. However, you need to access apps when using `pull_request_bypassers`. Public apps should work, but if you reference an internal one, it will fail when using app authentication. App authentication for the provider does not make this possible; see #1248.


Sad times to use the provider :-(

**Comment by kfcampbell** [comment]
@0x46616c6b if you have ideas about how we can handle this better, we're very accepting of new issues and PRs!

**Comment by 0x46616c6b** [comment]
Would it be possible to remove the deprecated field `contexts` from the branch protection v3? How the actual policy for removing a deprecated field/feature looks in the provider right now?

**Comment by kfcampbell** [comment]
I think it's reasonable to do a major version release in the near future! It's been quite a while since we last released a breaking change ([September 14th](https://github.com/integrations/terraform-provider-github/releases/tag/v5.0.0)). There's a few other changes I wouldn't mind batching in there as well, such as #1029, #448, and maybe even #1780. 

In general, Hashicorp recommends not doing more than one breaking change a year if we can help it.

**Comment by entropitor** [comment]
Any progress on this? This kinda makes github_branch_protection_v3 unusable.

**Comment by xiaoyanli-lyft** [comment]
@kfcampbell any timeline we have to officially remove contexts from the branch protection v3? this issue had been on for a while

**Comment by kfcampbell** [comment]
Unfortunately GitHub's SDK team doesn't have the bandwidth to work on this ourselves right now, though PRs are appreciated!

**Comment by yorik** [comment]
We have several issues related to that:
```
  ~ resource "github_branch_protection_v3" "this" {
        id                              = "foo:master"
      ~ require_signed_commits          = false -> true
        # (5 unchanged attributes hidden)

      ~ required_status_checks {
          ~ checks         = [
              - "foo",
              + "foo:-1",
            ]
          ~ contexts       = [
              - "foo",
            ]
            # (2 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }

  ~ resource "github_branch_protection_v3" "this" {
        id                              = "bar:main"
        # (6 unchanged attributes hidden)

      ~ required_status_checks {
          ~ contexts       = [
              - "TruffleHog",
            ]
            # (3 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }
```

So there should be added special check for `app_id == -1` and probably it's possible to hack that if checks do exist in terraform state just ignore contexts completely.

Would that work?

**Comment by yorik** [comment]
This is still very annoying issue: it shows huge diff, while actual change could be small, so it's easy to miss something.

**Comment by rsi-mrobinson** [comment]
Also facing this, tried adding the `contexts` to my terraform as I was not using the field since it's been marked as `deprecated`. So instead of useless plan output showing changes that don't matter, now I have `GitHub is deprecating the use of `contexts`. Use a `checks` array instead.` yippee...

**Comment by rsi-mrobinson** [comment]
I love that #2232 potentially fixes this issue and it's been sitting open and unreviewed since APRIL.

**Comment by stevehipwell** [comment]
I think this issue has been caused by the lack of having `Computed: true` set for `contexts` & `checks`. I'm working through the acceptance tests in #2476 and have fixed this by adding `Computed`.

**Comment by bahag-klickst** [comment]
As this issue looks like not getting resolved any time soon, we decided to migrate all our branch protection rules to the graphql based branch protections
https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection

**Comment by stevehipwell** [comment]
This should have been fixed by #2502.
